### PR TITLE
Forgotten password

### PR DESCRIPTION
**Pull request by pedbr**
Minor refactoring on Steps files
Adds tests for Forgotten Password feature
PT> https://www.pivotaltracker.com/story/show/167062851

### CODE REVIEW

File: features/forgotten_password.feature [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Feature: Forgotten Password
+    As a user,
+    In order to help me remember my CA Mailboxer Account Password,
+    I would like to have a "Forgot Password" option to help me retrieve my forgotten password
+
+Background:
```
Comment (oliverochman) [inline features/forgotten_password.feature]: please correct the indentation in this file

File: features/forgotten_password.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    Given the following user exists
+    | email        | password   | name |
+    | john@doe.com | password1  | John |
+    And I visit the site
+    And I click "Login"
+    And I click "Forgot your password?"
+    And I should see "Forgot your password?"
```
Comment (oliverochman) [inline features/forgotten_password.feature]: never have assertions in the background

File: features/forgotten_password.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Scenario: If user forgets password they can request it to be recovered (happy path)
+    And I fill "Email" with "john@doe.com"
```
Comment (oliverochman) [inline features/forgotten_password.feature]: Change to `When I fill "Email" with "john@doe.com"`

File: features/forgotten_password.feature [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    And I click "Send me reset password instructions"
+    Then I should see "You will receive an email with instructions on how to reset your password in a few minutes."
+
+Scenario: Not possible to recover password due to invalid email (sad path)
+    And I fill "Email" with "john"
```
Comment (oliverochman) [inline features/forgotten_password.feature]: Same comment as line 16

File: features/forgotten_password.feature [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    And I click "Send me reset password instructions"
+    Then I should see "Email not found"
+
+Scenario: Not possible to recover password due no user with that email (sad path)
+    And I fill "Email" with "michael@doe.com"
```
Comment (oliverochman) [inline features/forgotten_password.feature]: Same comment as line 16

File: features/forgotten_password.feature (hunk 1 above)
Comment (sethnejame) [reply features/forgotten_password.feature]: done

File: features/forgotten_password.feature (hunk 2 above)
Comment (sethnejame) [reply features/forgotten_password.feature]: done

File: features/forgotten_password.feature (hunk 3 above)
Comment (sethnejame) [reply features/forgotten_password.feature]: done

File: features/forgotten_password.feature (hunk 4 above)
Comment (sethnejame) [reply features/forgotten_password.feature]: done

File: features/forgotten_password.feature (hunk 5 above)
Comment (sethnejame) [reply features/forgotten_password.feature]: done

**Comment by sethnejame** [comment]
Everything's fixed and pushed up!

File: features/step_definitions/basic_steps.rb [hunk 6]
```diff
@@ -1,13 +1,29 @@
...
   table.hashes.each do |user|        
-       FactoryBot.create(:user, user)
+      FactoryBot.create(:user, user)
     end
   end
 
+Given("the following messages exist") do |table|
+  table.hashes.each do |email|
+    sender = User.find_by(name: email[:sender])
+    @receiver = User.find_by(name: email[:receiver])
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb:10]: do you need to use an instance variable here?

File: features/forgotten_password.feature [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    Background:
+        Given the following user exists
+        | email        | password   | name |
+        | john@doe.com | password1  | John |
+        And I visit the site
+
+    Scenario: If user forgets password they can request it to be recovered (happy path)
+        When I click "Login"
+        And I click "Forgot your password?"
+        And I should see "Forgot your password?"
```
Comment (oliverochman) [inline features/forgotten_password.feature]: Please change to `Then I should see "Forgot your password?"`

Comment (sethnejame) [reply features/forgotten_password.feature]: done

File: features/step_definitions/basic_steps.rb (hunk 6 above)
Comment (sethnejame) [reply features/step_definitions/basic_steps.rb:10]: This functionality was approved on another pull request for the Delete testing

**Comment by sethnejame** [comment]
Done

**Pull request closed without merging**